### ISSUE TITLE
Seeding data with double values can lose precision

### ISSUE THREAD

**Issue by edwiles** [dotnet/efcore#15980]
If HasData is used to seed fields of type double, the generated migration often loses precision in the values.

### Steps to reproduce

1. Create an entity with a double property:

```C#
public class MyEntity
{
    public int Id { get; set; }
    public double Data { get; set; }
}
```

2. Seed the data during OnModelCreating:

```C#
    modelBuilder.Entity<MyEntity>().HasData(new[]
        {
            new MyEntity { Id = 1, Data = 0.00404686 },
            new MyEntity { Id = 2, Data = 79.8978 },
        });
```

3. Add an initial migration:

```C#
dotnet ef migrations add InitialCreate
```

The generated migration seeds the data but uses imprecise values:

```
            migrationBuilder.InsertData(
                table: "MyEntities",
                columns: new[] { "Id", "Data" },
                values: new object[] { 1, 0.0040468600000000002 });

            migrationBuilder.InsertData(
                table: "MyEntities",
                columns: new[] { "Id", "Data" },
                values: new object[] { 2, 79.897800000000004 });
```

4. Change "double" to "float" in MyEntity, and put an "f" beside the floating-point values in HasData. This results in precise floats being generated.

For reference, our system's first migration contains the doubles precisely and it was generated using version 2.1.1.

### Further technical details
EF Core version: 2.2.4
.NET SDK: 2.2.104
Database Provider: Microsoft.EntityFrameworkCore.Sqlite (though in our "real" code we use SqlServer)
Operating system: Windows 10 Enterprise
IDE: Visual Studio 15.9.7

**Comment by ajcvickers** [comment]
Related to #14952

**Comment by ajcvickers** [comment]
@edwiles We discussed this in triage and we don't believe there is any error here. The nature of floating point representations means that values often cannot be represented exactly. This means that `0.00404686` as a C# literal is exactly the same value as `0.0040468600000000002`. (This is pretty easy to check.)

The change we made from 2.1 is to avoid other cases of data loss by always writing the literals out with full precision.
